### ISSUE TITLE
Custom Config

### ISSUE THREAD

**Issue by techcto** [aws-quickstart/taskcat#567]
In the new Taskcat, can you pass a custom config?  This feature was helpful for forking another repo and not altering the original Taskcat config.

**Comment by techcto** [comment]
Other feature requests:

If taskcat fails it does not cleanup S3 bucket.  Sometimes this is good, sometimes bad.  If your not careful you will hit AWS S3 bucket limits. An option to possibly clean failed test buckets would be nice. Especially since deleting S3 buckets now requires them to be empty first. I have tried the clean function with not much luck.  Maybe user error?

Ability to run a specific test vs. all tests

Not sure if this is also project specific, but ability to run test and only upload the single cloudformation file referenced vs. whole project. And also not enable build_submodules for a single test. The use-case here is an add-on cloudformation template.

**Comment by andrew-glenn** [comment]
Hey @techcto , thanks for your feedback. 

### Custom Config

We currently support a custom config file. You can pass it in with the `-i` flag.

```
❯ taskcat test run -p /path/to/project -i /path/to/other-taskcat-config.yml
 _            _             _
| |_ __ _ ___| | _____ __ _| |_
| __/ _` / __| |/ / __/ _` | __|
| || (_| \__ \   < (_| (_| | |_
 \__\__,_|___/_|\_\___\__,_|\__|

```
We also support specific tests vs all tests. This was added by Jay McConnell in PR #525. It'll be included in `0.9.18`. If you'd like it in the interim, we build development releases.

`pip install taskcat==0.9.18.dev7`

### S3 Buckets

For S3 Buckets. It's a bit more nuanced. We have a couple of patterns for S3 buckets. The first is one-bucket-per-test. We cleanup those buckets (or should be). 

The second pattern is regional buckets. This creates an S3 bucket `taskcat-<ACCOUNT_HASH>-<REGION>`. The intent is to re-use the bucket to cut down on account limits. We currently leave those as-is. This is enabled through the `s3_regional_buckets` config option.

The flip side though, is that only changed files will be uploaded with your next test, since the bucket is populated with most of the content you need. 

If what you're seeing doesn't match what I've described, then we need to dig in, because that isn't intended state.

**Comment by andrew-glenn** [comment]
Housekeeping: no traction on this issue in some time; closing.
